### PR TITLE
mime: add tests for bug 6207

### PR DESCRIPTION
**Pull request by inashivb**
## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/6207

Previous PR: #1312 

Changes since v4:
- Add `file.data` rule to match on file content for test `bug-6207-2`

### CODE REVIEW

**Review by catenacyber — APPROVED** [review]
Good :-)

File: tests/bug-6207-2/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Test for the edge case that should be handled properly by MIME decoder while
+following RFC2045.
+
+```
+NA=
+=Mg
+==
+```
+should ideally get decoded to `42` as demonstrated in this test.
```
Comment (catenacyber) [inline tests/bug-6207-2/README.md:11]: as `NA==Mg==` gets decoded into `42` as well (whatever the line feeds)

File: tests/bug-6207-2/invalid-base64-mime.syn [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+default < (content:"220 smtp.server.com ESMTP Postfix\x0d\x0a";);
+default > (content:"EHLO smtp.intra\x0d\x0a";);
+default < (content:"250-smtp.lab.com\x0d\x0a250-PIPELINING\x0d\x0a250-SIZE 10240000\x0d\x0a250-VRFY\x0d\x0a250-ETRN\x0d\x0a250-STARTTLS\x0d\x0a250-ENHANCEDSTATUSCODES\x0d\x0a250-8BITMIME\x0d\x0a250-DSN\x0d\x0a250-SMTPUTF8\x0d\x0a250 CHUNKING\x0d\x0a";);
+default > (content:"MAIL FROM:blah@smtp.lab.com\x0d\x0a";);
+default < (content:"250 2.1.0 Ok\x0d\x0a";);
+default > (content:"RCPT TO:test@wut.com\x0d\x0a";);
+default < (content:"250 2.1.5 Ok\x0d\x0a";);
+default > (content:"DATA\x0d\x0a";);
+default < (content:"354 End data with <CR><LF>.<CR><LF>\x0d\x0a";);
+default > (content:"Subject: SMTPbelka-test_sans_name2021-03-08-17:28:53-221a0d8d17b3b41e28ec113dcabb55da7bdb03a8c0bb5d3de252f5d69347aa4d.zip\x0d\x0a";);
```
Comment (catenacyber) [inline tests/bug-6207-2/invalid-base64-mime.syn:11]: We could change the dummy subject

File: tests/bug-6207-2/invalid-base64-mime.syn [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+default > (content:"\x0d\x0a";);
+default > (content:"--KkK170891tpbkKk__FV_KKKkkkjjwq\x0d\x0a";);
+default > (content:"Content-Type: text/plain; charset=UTF-8; format=flowed\x0d\x0a";);
+default > (content:"Content-Disposition: inline\x0d\x0a";);
+default > (content:"\x0d\x0a";);
+default > (content:"Ceci est un test\x0d\x0a";);
+default > (content:"\x0d\x0a";);
+default > (content:"\x0d\x0a";);
+default > (content:"--KkK170891tpbkKk__FV_KKKkkkjjwq\x0d\x0a";);
+default > (content:"Content-Type: application/zip;\x0d\x0a";);
```
Comment (catenacyber) [inline tests/bug-6207-2/invalid-base64-mime.syn:25]: This is no longer a zip ;-)

File: tests/bug-6207-2/suricata.rules [hunk 4]
```diff
@@ -0,0 +1 @@
+alert tcp any any -> any any (msg: "Test file content"; file.data; content:"42"; sid:1;)
```
Comment (catenacyber) [inline tests/bug-6207-2/suricata.rules:1]: Can we add `filesize: 2` keyword to ensure the file is exactly 42 ant not having 42 as a substring ?

**Comment by victorjulien** [comment]
Merged in #1315, thanks!
